Update Xdebug to v3.1.5 whenever possible

jrfnl opened the pull request:
Released June 6, 2022.

Changelog: https://xdebug.org/updates

I noticed that PR #61, which added Xdebug 3 to the PHP 8 image (and subsequent PRs #68 and #78) only added Xdebug 3.x to the PHP 8.x images, while Xdebug 3 is available for PHP 7.2, 7.3 and 7.4 as well.

I propose to always use the latest Xdebug available for a particular PHP version, so this PR includes updating the Xdebug version for PHP 7.2, 7.3 and 7.4 to Xdebug 3.1.5.

Note: this may mean that the Xdebug related directive in `php.ini` files need to be adjusted, but I couldn't find where to do so.
Also see: https://xdebug.org/docs/upgrade_guide

jrfnl committed:
Update Xdebug to v3.1.5 whenever possible

Released June 6, 2022.

Changelog: https://xdebug.org/updates

I noticed that PR 61 which added Xdebug 3 to the PHP 8 image (and subsequent PRs 68 and 78) only added Xdebug 3.x to the PHP 8.x images, while Xdebug 3 _is_ available for PHP 7.2, 7.3 and 7.4 as well.

I propose to always use the latest Xdebug available for a particular PHP version, so this PR includes updating the Xdebug version for PHP 7.2, 7.3 and 7.4 to Xdebug 3.1.5.

Note: this may mean that the Xdebug related directive in `php.ini` files need to be adjusted, but I couldn't find where to do so.
Also see: https://xdebug.org/docs/upgrade_guide

desrosj reviewed (approved):
Looks good. Thanks @jrfnl.

I think that I fixed the build issues in #84. Could you merge the latest from `master` just to verify everything builds correctly?

jrfnl commented:
> I think that I fixed the build issues in #84. Could you merge the latest from `master` just to verify everything builds correctly?

Yes, that weird build failure was outside of my scope. I have rebased all three of my open PRs now and expect them all to pass (if not, I'll look into it)
